mbed_lstools/lstools_base.py

```diff
@@ -73,6 +73,8 @@ def __init__(self):
         self.lock_file = os.path.join(MbedLsToolsBase.HOME_DIR,
                                       MbedLsToolsBase.MBEDLS_HOME_DIR, MbedLsToolsBase.MBEDLS_GLOBAL_LOCK)
         self.mbedls_get_mocks()
+        # Make sure retargeting is applied if retarget file exists
+        self.retarget(True)
```

Comment:
@mazimkhan @bridadan Considering this PR has already taken more of my time then I expected it to: if this doesn't satisfy the mbed team, can I rely on you fixing the underlying issue one way or another? It really wasn't my duty to fix it to begin with...

Retargeting needs to work in all mbed tools. The way they currently interface with mbed-ls doesn't apply retargeting. Can you please fix this ASAP so us silicon partners can continue testing our future targets?